components/locid/src/langid.rs

```diff
@@ -142,6 +142,18 @@ impl LanguageIdentifier {
     }
 }
 
+impl AsRef<LanguageIdentifier> for LanguageIdentifier {
+    fn as_ref(&self) -> &LanguageIdentifier {
+        self
+    }
+}
+
+impl AsMut<LanguageIdentifier> for LanguageIdentifier {
+    fn as_mut(&mut self) -> &mut LanguageIdentifier {
+        self
+    }
+}
+
```

Comment:
Why do you need that?

Reply:
I agree it really doesn't look like this should be necessary, but I hit errors, for instance in the LocaleCanonicalizer::minimize function if I remove it. That might just mean I have a problem with how I've defined things, but I'm having a hard time tracking it down. Does anything look suspicious to you?

Reply:
This was discussed on slack, it looks like Borrow and BorrowMut have default implementations that would avoid this code, but those are only usable if hash and equality are defined the same way for LID and the LID wrapped by Locale. As far as I can tell, this is the case and we can use the Borrow variants here, but part of me thinks that these two extra definitions are a small price to pay to not have to worry about this. Not a strongly held opinion, please let me know which way you'd like to go with this.